From d742e82240598e8ba86ee4e97ca6980bf9f81328 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:46 -0400
Subject: [ethernet] bnx2x: remove null test before kfree

Message-id: <1409946615-5862-42-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91582
O-Subject: [RHEL7.1 PATCH 41/70] drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c: remove null test before kfree
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 88729dd18995a87f0f8fe0423518b171a877d7b6
Author: Fabian Frederick <fabf@skynet.be>
Date:   Tue Jun 17 21:32:53 2014 +0200

    drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c: remove null test before kfree

    Fix checkpatch warning:
    WARNING: kfree(NULL) is safe this check is probably not required

    Cc: Ariel Elior <ariele@broadcom.com>
    Cc: netdev@vger.kernel.org
    Signed-off-by: Fabian Frederick <fabf@skynet.be>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c
index adecc1d..8b7150e 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c
@@ -597,8 +597,7 @@ int bnx2x_vf_mcast(struct bnx2x *bp, struct bnx2x_virtf *vf,
  rc = bnx2x_config_mcast(bp, &mcast, BNX2X_MCAST_CMD_DEL);
  if (rc) {
   BNX2X_ERR("Failed to remove multicasts\n");
-  if (mc)
-   kfree(mc);
+  kfree(mc);
   return rc;
  }
 
-- 
1.7.1